stormpath/stormpath-sdk-react
# Fix: React unknown prop

Fixes the React "unknown prop" error present when using the latest React version (see issue #95).
#### How to verify
1. Checkout the `fix-react-unknown-prop` branch of the [React example application](https://github.com/stormpath/stormpath-express-react-example).
2. Checkout this branch.
3. Run `$ npm run build-cjs`.
4. Link the example application to this branch.
5. Open the terminal, go to the example application and run `$ npm start`.
6. Open up your browser and navigate to `http://localhost:3000/login`.
7. Open up the console and verify that there are no errors.
8. Login using the username/password login form.
9. Verify that you are logged in (redirected to `/profile`) and that there are no errors in the console.
10. Navigate around the example application and verify that there are no errors in the console.
#### Additional

Since we cannot use the `spIf` and `spBind` attributes directly on certain elements anymore without receiving warnings, this PR fixes the issue by allowing the attributes to be prefixed with `data-`. I.e. using `data-spIf` instead of `spIf` and `data-spBind` instead of `spBind` is now recommended instead. In the future we might move to using components (`If` and `Bind`) but that will require quite some work, so meantime we'll just fix it by prefixing.

Fixes #95
